x-pack/dockerlogbeat/pipelinemanager/pipelineManager.go

```diff
@@ -103,12 +93,26 @@ func (pm *PipelineManager) CreateClientWithConfig(logOptsConfig map[string]strin
 //===================
 // Private methods
 
-// getPipeline gets a pipeline based on a confighash
-func (pm *PipelineManager) getPipeline(hashstring string) (*Pipeline, bool) {
+// checkAndCreatePipeline performs the pipeline check and creation as one atomic operation
+// It will either return a new pipeline, or an existing one from the pipeline map
+func (pm *PipelineManager) checkAndCreatePipeline(logOptsConfig map[string]string, file string) (*Pipeline, error) {
 	pm.mu.Lock()
 	defer pm.mu.Unlock()
-	pipeline, exists := pm.pipelines[hashstring]
-	return pipeline, exists
+
+	hashstring := makeConfigHash(logOptsConfig)
+
+	var pipeline *Pipeline
+	var err error
+	pipeline, test := pm.pipelines[hashstring]
```

Comment:
When using a hash as key I'm always a little woried about false-positives due to hash collisions. Have you considered to compute a key value instead of using a hash value?

Reply:
Added this to #13990